dvc/output/base.py

```diff
@@ -90,6 +92,8 @@ def __init__(
         # should be absolute and don't contain remote:// refs.
         self.stage = stage
         self.repo = stage.repo if stage else None
+        if self.repo is not None and path_isin(path, self.repo.root_dir):
+            path = os.path.relpath(path, self.repo.root_dir)
```

Comment:
A test would be nice too 🙂

Reply:
@casperdcl Oh, one more thing. We should move this check down below, below `self.path_info = self._parse_path(remote, path)`, as there are some parcing nuances. You will be able to use `self.path_info.isin` too btw.

Reply:
Oh, btw, we need to move this check down below, below `self.path_info = self._parse_path(remote, path)` as there are some parsing nuances.

Reply:
Plus, note that we are modifying `OutputBase` but our check should only work for `local`. So let's move to `OutputLOCAL.__init__`

Reply:
done